akka-remote/src/test/scala/akka/remote/transport/SystemMessageDeliveryStressTest.scala

```diff
@@ -97,36 +114,64 @@ abstract class SystemMessageDeliveryStressTest(msg: String, cfg: String)
+  val addressA = systemA.asInstanceOf[ExtendedActorSystem].provider.getDefaultAddress
   val addressB = systemB.asInstanceOf[ExtendedActorSystem].provider.getDefaultAddress
 
-  val root = RootActorPath(address)
   // We test internals here (system message delivery) so we are allowed to cheat
-  val there = RARP(systemB).provider.resolveActorRef(root / "temp" / sysMsgVerifier.path.name).asInstanceOf[InternalActorRef]
+  val targetForA = RARP(systemA).provider.resolveActorRef(RootActorPath(addressB) / "temp" / sysMsgVerifierB.path.name)
+  val targetForB = RARP(systemB).provider.resolveActorRef(RootActorPath(addressA) / "temp" / sysMsgVerifierA.path.name)
 
   override def atStartup() = {
-    system.eventStream.publish(TestEvent.Mute(
+    systemA.eventStream.publish(TestEvent.Mute(
+      EventFilter[EndpointException](),
       EventFilter.error(start = "AssociationError"),
-      EventFilter.warning(pattern = "received dead letter.*")))
+      EventFilter.warning(pattern = "received dead .*")))
     systemB.eventStream.publish(TestEvent.Mute(
       EventFilter[EndpointException](),
       EventFilter.error(start = "AssociationError"),
-      EventFilter.warning(pattern = "received dead letter.*")))
+      EventFilter.warning(pattern = "received dead .*")))
+
+    systemA.eventStream.subscribe(probeA.ref, classOf[QuarantinedEvent])
+    systemB.eventStream.subscribe(probeB.ref, classOf[QuarantinedEvent])
   }
 
   "Remoting " + msg must {
     "guaranteed delivery and message ordering despite packet loss " taggedAs TimingTest in {
-      Await.result(RARP(systemB).provider.transport.managementCommand(One(address, Drop(0.3, 0.3))), 3.seconds.dilated)
-      systemB.actorOf(Props(classOf[SystemMessageSender], MsgCount, there))
+      import systemA.dispatcher
+
+      val transportA = RARP(systemA).provider.transport
+      val transportB = RARP(systemB).provider.transport
 
-      val toSend = (0 until MsgCount).toList
-      val received = expectMsgAllOf(45.seconds, toSend: _*)
+      Await.result(transportA.managementCommand(One(addressB, Drop(0.2, 0.2))), 3.seconds.dilated)
+      Await.result(transportB.managementCommand(One(addressA, Drop(0.2, 0.2))), 3.seconds.dilated)
+
+      // Schedule peridodic disassociates
+      systemA.scheduler.schedule(1.second, 3.seconds) {
+        transportA.managementCommand(ForceDisassociateExplicitly(addressB, reason = AssociationHandle.Unknown))
+      }
+
+      systemB.scheduler.schedule(2.seconds, 3.seconds) {
+        transportB.managementCommand(ForceDisassociateExplicitly(addressA, reason = AssociationHandle.Unknown))
+      }
+
+      systemB.actorOf(Props(classOf[SystemMessageSender], msgCount, burstSize, burstDelay, targetForB))
+      systemA.actorOf(Props(classOf[SystemMessageSender], msgCount, burstSize, burstDelay, targetForA))
+
+      val toSend = (0 until msgCount).toList
+      for (m ← 0 until msgCount) {
+        probeB.expectMsg(30.seconds, m)
+        probeA.expectMsg(30.seconds, m)
```

Comment:
can each msg really take 30 seconds? should there be a `within` block for the total timeout and something smaller for each message?